doc/encoding.rdoc

```diff
@@ -132,7 +132,232 @@ returns the \Encoding of the concatenated string, or +nil+ if incompatible:
+  sym.encoding # => #<Encoding:US-ASCII>
+  re = /foo/   # => /foo/
+  re.encoding  # => #<Encoding:US-ASCII>
+
+=== Filesystem \Encoding
+
+The filesystem encoding is the default \Encoding for a string from the filesystem:
+
+    Encoding.find("filesystem") # => #<Encoding:UTF-8>
+
+=== Locale \Encoding
+
+The locale encoding is the default encoding for a string from the environment,
+other than from the filesystem:
+
+  Encoding.find('locale') # => #<Encoding:IBM437>
+
+=== \IO Encodings
+
+An IO object (an input/output stream), and by inheritance a File object,
+has at least one, and sometimes two, encodings:
+
+- Its _external_ _encoding_ identifies the encoding of the stream.
+- Its _internal_ _encoding_, if not +nil+, specifies the encoding
+  to be used for the string constructed from the stream.
+
+==== External \Encoding
+
+Bytes read from the stream are decoded into characters via the external encoding;
+by default (that is, if the internal encoding is +nil),
+those characters become a string whose encoding is set to the external encoding.
+
+The default external encoding is:
+
+- UTF-8 for a text stream:
+
+    # File.
+    f = File.open('t.txt', 'rt')
+    f.external_encoding  # => #<Encoding:UTF-8>
+
+    # IO.
+    fd = IO.sysopen('t.txt', 'rt')
+    io = IO.new(fd)
+    io.external_encoding # => #<Encoding:UTF-8>
+
+- ASCII-8BIT for a binary stream:
+
+    f = File.open('t.rus', 'rb')
+    f.external_encoding  # => #<Encoding:ASCII-8BIT>
+
+    # IO.
+    fd = IO.sysopen('t.rus', 'rb')
+    io = IO.new(fd)
+    io.external_encoding # => #<Encoding:UTF-8>
+
+The external encoding may be set by the open option +external_encoding+:
+
+  # File.
+  f = File.open('t.txt', external_encoding: 'ASCII-8BIT')
+  f.external_encoding                               # => #<Encoding:ASCII-8BIT>
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd, external_encoding: 'ASCII-8BIT')
+  io.external_encoding # => #<Encoding:ASCII-8BIT>
+
+The external encoding may also set by method #set_encoding:
+
+  # File.
+  f = File.open('t.txt')
+  f.set_encoding('ASCII-8BIT')
+  f.external_encoding  # => #<Encoding:ASCII-8BIT>
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd)
+  io.set_encoding('ASCII-8BIT')
+  io.external_encoding # => #<Encoding:ASCII-8BIT>
+
+==== Internal \Encoding
+
+If not +nil+, the internal encoding specifies that the characters read
+from the stream are to be converted to characters in the internal encoding;
+those characters become a string whose encoding is set to the internal encoding.
+
+The default internal encoding is +nil+ (no conversion):
+
+  # File.
+  f = File.open('t.txt')
+  f.internal_encoding # => nil
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd)
+  io.internal_encoding # => nil
+
+The internal encoding may set by the open option +internal_encoding+:
+
+  # File.
+  f = File.open('t.txt', internal_encoding: 'ASCII-8BIT')
+  f.internal_encoding  # => #<Encoding:ASCII-8BIT>
+
+  # IO.
```

Comment:
As above, let's keep the File example, and remove the IO example.

Reply:
Done.